Assets/Mirage/Weaver/Processors/SyncVarProcessor.cs

```diff
@@ -413,7 +413,15 @@ void WriteEndFunctionCall()
             {
                 // only use Callvirt when not static
                 OpCode opcode = hookMethod.IsStatic ? OpCodes.Call : OpCodes.Callvirt;
-                worker.Append(worker.Create(opcode, hookMethod));
+                MethodReference hookMethodReference = hookMethod;
+                // If the delcaring type is generic we need to duplicate the method reference.
```

Comment:
as a rule of thumb,  don't explain what this is doing.   Explain why instead.

I can see that you are duplicating the method ref just by reading the code,  the comment is redundant.  What I can't figure out is why we are doing it.

Reply:
I removed duplicate and changed it with some more straightforward code. I hope it becomes more clear. 
Removed the duplicate in 86a8205 and added a comment in cb40e99.